internal/consensus/state.go

```diff
@@ -249,7 +249,7 @@ func (cs *State) GetLastHeight() int64 {
 
 // GetRoundState returns a shallow copy of the internal consensus state.
 // This function is thread-safe.
-func (cs *State) GetRoundState() *cstypes.RoundState {
+func (cs *State) GetRoundState() cstypes.RoundState {
```

Comment:
This is a breaking change. But does it need to be one? We can just change the private method instead.

Reply:
Nothing wrong with breaking change here. We are not going to backport this, don't we?

Reply:
We are probably not, but I am not really convinced about the performance improvement that it can provide.

So, if we are to break, lets do that while providing a measurable benefit.

Reply:
I can collect profiling on this change today and follow up w a comment. Removal of the hot loop `updateRoundStateRoutine()` was enough reason for me to tackle this change without a profile, as removing a Go routine that always copies is almost definitely better, but I can provide numbers to back up.